apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -1,330 +1,520 @@
-  uploadSpriteToAnimationLibrary,
+  uploadAnimationToAnimationLibrary,
   uploadMetadataToAnimationLibrary
 } from '@cdo/apps/assetManagement/animationLibraryApi';
 
-const SpriteLocation = makeEnum('library', 'level');
-const UploadStatus = makeEnum(
-  'fail',
-  'success',
-  'filenameOverride',
-  'badFilename',
-  'none'
-);
+const EXTENSION_CHECK = 'extensionError';
+const FILENAME_CHECK = 'filenameError';
+const FILESIZE_CHECK = 'filesizeError';
+const ANIMATION_EXISTS_CHECK = 'animationExistsError';
+const AnimationLocations = makeEnum('library', 'level');
 
-// Levelbuilder tool for adding sprites to the Spritelab animation library.
-export default class SpriteUpload extends React.Component {
+const initialSpriteLabLibrariesState = {
+  spriteLabLevelAnimations: [],
+  spriteLabLibraryCategories: [],
+  spriteLabLibraryAnimations: []
+};
+
+const newImageState = {
+  fileData: null,
+  filename: '',
+  filePreviewURL: ''
+};
+
+const imageChecksInitialState = {
+  [EXTENSION_CHECK]: Status.WAITING,
+  [FILENAME_CHECK]: Status.WAITING,
+  [FILESIZE_CHECK]: Status.WAITING,
+  [ANIMATION_EXISTS_CHECK]: Status.WAITING
+};
+
+const initialUserInputMetaData = {
+  availability: '',
+  category: '',
+  aliases: []
+};
+
+const initialGeneratedMetadata = {
+  frameSize: {x: NaN, y: NaN},
+  destination: '',
+  metadata: ''
+};
+
+const initialComponentState = {
+  ...newImageState,
+  ...imageChecksInitialState,
+  ...initialGeneratedMetadata
+};
+
+const maxSize = 800;
+const minSize = 100;
+
+// Levelbuilder tool for adding animations to the Spritelab animation library.
+export default class AnimationUpload extends React.Component {
   state = {
-    fileData: null,
-    filePreviewURL: '',
-    filename: '',
-    spriteAvailability: '',
-    category: '',
-    currentCategories: [],
-    currentLibrarySprites: [],
-    currentLevelSprites: [],
-    aliases: [],
-    metadata: '',
-    uploadStatus: {
-      status: UploadStatus.none,
-      message: ''
-    }
+    ...initialSpriteLabLibrariesState,
+    ...initialUserInputMetaData,
+    ...initialComponentState,
+    uploadStatus: Status.WAITING,
+    errorMessage: ''
   };
-
   componentDidMount() {
-    // Get list of sprites from level-specific folder
+    // Get list of animations from level-specific folder
     getLevelAnimationsFiles().then(files => {
-      this.setState({currentLevelSprites: Object.keys(files)});
+      this.setState({spriteLabLevelAnimations: Object.keys(files)});
     });
 
-    // Get data from the spritelab library manifest
+    // Get data from the spritelab animation library manifest
     getManifest('spritelab', 'en_us').then(data => {
       this.setState({
-        currentCategories: Object.keys(data.categories),
-        currentLibrarySprites: Object.keys(data.metadata)
+        spriteLabLibraryCategories: Object.keys(data.categories),
+        spriteLabLibraryAnimations: Object.keys(data.metadata)
       });
     });
   }
 
   handleSubmit = event => {
```

Comment:
A more descriptive name (or comment) might help readability for what exactly is being submitted.